### PR TITLE
server admin API definition

### PR DESCRIPTION
**Pull request by wibarre**
Signed-off-by: William Barrera <wibarre@gmail.com>

### CODE REVIEW

File: api/v1/api.yaml [hunk 1]
```diff
@@ -14,60 +14,128 @@ components:
...
-          format: bytes
-        createdAt:
-          type: string
-          format: date-time
         status:
           type: string
+          format: string
           enum:
-            - active
             - inactive
```
Comment (mchurichi) [inline api/v1/api.yaml:48]: what's the difference between inactive and disabled?

File: api/v1/api.yaml [hunk 2]
```diff
@@ -14,60 +14,128 @@ components:
...
+        - id
+        - trustdomain
+        - description
+        - status
     FederationGroupMembership:
       # A FederationGroupMembership reperesents a particular SPIRE server's presence in a 
       # FederationGroup
       type: object
       properties:
+        Id:
```
Comment (mchurichi) [inline api/v1/api.yaml:85]: nit: this should be in lowercase

File: api/v1/api.yaml [hunk 3]
```diff
@@ -219,7 +530,268 @@ paths:
...
+              schema:
+                $ref: '#/components/schemas/FederationGroup'
+        default:
+          description: unexpected error
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'  
+    put:
+      description: update an federationGroupMembership by ID
```
Comment (mchurichi) [inline api/v1/api.yaml:628]:
```suggestion
      description: update a federationGroupMembership by ID
```

File: api/v1/api.yaml [hunk 4]
```diff
@@ -219,7 +530,268 @@ paths:
...
+        '204':
+          description: no content
+        default:
+          description: unexpected error
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+    delete:
+      description: delete an federationGroup membership
```
Comment (mchurichi) [inline api/v1/api.yaml:654]:
```suggestion
      description: delete a federationGroup membership
```

**Review by n-th — APPROVED** [review]
🍿

File: pkg/server/api/v1/api_harvester.yaml [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+        spireServerConsent:
+          type: string
+          format: string
+        status:
+          type: string
+          enum:
+            - active
+            - inactive
+            - invited
+      required:
```
[the remark this answers is not in the document]
Comment (wibarre) [reply pkg/server/api/v1/api_harvester.yaml:44]: agreed.

File: pkg/server/api/v1/api_server.yaml [hunk 6]
```diff
@@ -0,0 +1,825 @@
...
+          type: string
+          format: string
+          enum:
+            - inactive
+            - active
+      required:
+        - id
+        - orgid
+        - name
+    #SpireServer is a representation of a trust-domain/SpireServer
```
[the remark this answers is not in the document]
Comment (wibarre) [reply pkg/server/api/v1/api_server.yaml:53]: agreed

File: pkg/server/api/v1/api_harvester.yaml [hunk 7]
```diff
@@ -0,0 +1,202 @@
+openapi: 3.0.0
+info:
+  title: SPIRE Bridge - Harverster API
```
[the remark this answers is not in the document]
Comment (wibarre) [reply pkg/server/api/v1/api_harvester.yaml]: add references to Galadriel that were missing

File: pkg/server/api/v1/api_harvester.yaml [hunk 8]
```diff
@@ -0,0 +1,202 @@
...
+    # Deletion is normally automatic (not through API) when the trust bundle contents
+    # indicate that it has expired
+    FederationRelationship:
+        # It is a representation of the federated relationships of a SPIRE server
+      type: object
+      properties:
+        id:
+          type: integer
+          format: int64
+        federationGroupId:
```
[the remark this answers is not in the document]
Comment (wibarre) [reply pkg/server/api/v1/api_harvester.yaml:19]: the id or the org and the federationGroup id will be the fields that will give us the possibility to create multiple "bridges" per organization.  Yes.

**Review by jufantozzi — APPROVED** [review]
💯

File: pkg/server/api/v1/api_harvester.yaml (hunk 8 above)
[the remark this answers is not in the document]
Comment (jufantozzi) [reply pkg/server/api/v1/api_harvester.yaml:19]: Ok, thank you 😄

**Pull request merged**